src/apps/metastreet/ethereum/metastreet.lending-v2.contract-position-fetcher.ts

```diff
@@ -0,0 +1,362 @@
+    const contractPositions = await this.getPositionsForBalances();
+
+    /* Filter out positions based on account's active positions */
+    const filteredPositions = contractPositions.filter(cp => ticks.has(cp.dataProps.positionKey));
+
+    const balances = await Promise.all(
+      filteredPositions.map(async contractPosition => {
+        const contract = multicall.wrap(this.getContract(contractPosition.address));
+        const balancesRaw = await this.getTokenBalancesPerPosition({ address, contract, contractPosition, multicall });
+        const allTokens = contractPosition.tokens.map((cp, idx) =>
+          drillBalance(cp, balancesRaw[idx]?.toString() ?? '0', { isDebt: cp.metaType === MetaType.BORROWED }),
+        );
+
+        const tokens = allTokens.filter(v => Math.abs(v.balanceUSD) > 0.01);
+        const balanceUSD = sumBy(tokens, t => t.balanceUSD);
+
+        const balance = { ...contractPosition, tokens, balanceUSD };
+        return balance;
+      }),
+    );
+    return balances;
+  }
+
+  async getTokenBalancesPerPosition(_params: GetTokenBalancesParams<Pool, DataProps>): Promise<BigNumberish[]> {
+    const contract: Pool = _params.contract;
+    const tick: BigNumber = _params.contractPosition.dataProps.tick;
+    const account: string = _params.address;
+    const implementationVersionMajor: string = _params.contractPosition.dataProps.implementationVersionMajor;
+
+    /* Get account's deposit logs and compute deposited amount and received shares */
+    const depositLogs = await contract.queryFilter(contract.filters.Deposited(account, tick), START_BLOCK_NUMBER);
+    const deposited: Deposited = depositLogs.reduce(
+      (deposited: Deposited, l) => {
+        if (l.args.tick.eq(tick) && l.args.account.toLowerCase() === account) {
+          return { amount: deposited.amount.add(l.args.amount), shares: deposited.shares.add(l.args.shares) };
+        } else {
+          return deposited;
+        }
+      },
+      { amount: constants.Zero, shares: constants.Zero },
+    );
+
+    /* Get account's withdrawal logs and compute withdrawn amount and burned shares */
+    let firstDepositBlockNumber: number = depositLogs.length > 0 ? depositLogs[0].blockNumber : START_BLOCK_NUMBER;
+    const withdrawLogs = await contract.queryFilter(contract.filters.Withdrawn(account, tick), firstDepositBlockNumber);
+    const withdrawn: Withdrawn = withdrawLogs.reduce(
+      (withdrawn: Withdrawn, l) => {
+        if (l.args.tick.eq(tick) && l.args.account.toLowerCase() === account) {
+          return { amount: withdrawn.amount.add(l.args.amount), shares: withdrawn.shares.add(l.args.shares) };
+        } else {
+          return withdrawn;
+        }
+      },
+      { amount: constants.Zero, shares: constants.Zero },
+    );
+
+    let redeemed: Redemption = { amount: constants.Zero, shares: constants.Zero };
+    /* Legacy implementation does not have latest deposits API */
+    if (implementationVersionMajor === '1') {
```

Comment:
It might be unclear from the code, but they are already separate positions because they have different contract addresses.

Reply:
I've noticed, but the reasons I think it would be a great idea to split it into 2 different positons are;
- They have different interfaces
- Since they don't have the same interfaces and method for fetching balances, the logic required to fetch balances is different based on the implementationVersion

Reply:
Got it